### PR TITLE
[SYCL] Ensure correct errc thrown for unassociated placeholder accessor (4.7.6.9)

### PR DESCRIPTION
**Pull request by cperkinsintel**
a simple fix for ensuring correct errc is thrown for an unassociated placeholder accessor (4.7.6.9)

Signed-off-by: Chris Perkins <chris.perkins@intel.com>

Note that this fix is not complete. It only addresses the most common cases.   There are still cases where an unassociated placholder accessor could get by undetected and crash in the kernel. Unfortunately, the fix for those cases is not trivial. I have the outlines of a fix, but it would be disruptive, too disruptive for such a low priority fix. I will open a ticket about it and discuss it with various stakeholders.

Expanded testing to cover this here: https://github.com/intel/llvm-test-suite/pull/890

### CODE REVIEW

**Comment by cperkinsintel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/890

**Comment by sergey-semenov** [comment]
@cperkinsintel It seems like this change can be covered by a unit test. Could you please add one?

**Comment by cperkinsintel** [comment]
@sergey-semenov  - I forgot to update the description. I already made a test for it here, this is also waiting for review: https://github.com/intel/llvm-test-suite/pull/890

**Comment by sergey-semenov** [comment]
Right, I meant that since handling this error case is limited to the DPCPP runtime component (i.e. does not require running any device code), a unit test would probably be a better way to verify this as opposed to a full on E2E test.

File: sycl/unittests/scheduler/ErrcUnboundPlaceholder.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
+//==-------- ErrcUnbouncPlaceholder.cpp ------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
```
Comment (cperkinsintel) [inline sycl/unittests/scheduler/ErrcUnboundPlaceholder.cpp:7]: @sergey-semenov  - can I get a consultation?

Getting back to this PR, I've added this unit test, and it works on Linux.  It has a very minimal mock.  But on Windows I'm getting a CL_INVALID_BINARY error.  At first I thought perhaps just calling `setupDefaultMockAPIs()` would fix it, but no such luck.

FWIW, I don't quite understand why redefining piDeviceGetInfo to be successful does the trick on Linux, but it does and the kernel is enqueued, the exception is thrown, the errc tested, etc. Awesome. 

But not on Windows. and I'm not sure what the best course is.  I can do tricks to get the unit test to succeed (like return PI_DEVICE_TYPE_ACC for any PI_DEVICE_INFO_TYPE request), but I think that ends up just with the test not actually running - though it's hard to say on Win.  

Also, FWIW, there seem to be environment dependencies that make working on these tests in isolation difficult (or make achieving repeatability difficult). But that's not the immediate concern.

**Review by steffenlarsen — APPROVED** [review]
Since the feature needs to report at the time of enqueuing a kernel, E2E tests should be included anyway. As such I am fine to not have a unittest for it.

**Comment by pvchupin** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/890

**Comment by cperkinsintel** [comment]
This got overlooked.  All tests are passing.  Needs merging.

**Pull request merged**